src/base_event.cpp

```diff
@@ -58,11 +58,22 @@ void MainFrame::OnLoadFiles(wxCommandEvent& event){
 }
 
 void MainFrame::enableGuiElements(bool inp){
-  // deactivate the Start and Load files buttons during the calculation
+  // deactivate the calculation and reload buttons during the calculation
   calcButton->Enable(inp);
   loadFilesButton->Enable(inp);
   // prevents editing of the atom list grid during the calculation
   atomListGrid->EnableEditing(inp);
+  // keeps calculation button deactivated if no valid structure is loaded
+  if (atomListGrid->GetNumberRows() == 0) {
+    calcButton->Enable(false);
+  }
+  // activates pdb specific options only if pdb file is loaded
+    if (getAtomFilepath().size() > 3 && getAtomFilepath().substr(getAtomFilepath().size()-4, 4) == ".pdb"){

```

Comment:
> I'd prefer you using the fileExtension() function in model_filereading.cpp. Doing it like this doesn't generalise if you have a file extension that isn't three characters long. 

Since this is specific to pdb files, I thought a simple line was better than calling a function to differentiate different file extensions.

> Also, why check the size of the extension? This is an unnecessary check. 

I probably should have clarified this. I first tested without this size check but when trying to load a filepath shorter than 4 characters (such as empty filepath), an error occurred because `str.size()-4` leads to a negative value while size_t type can't be negative.

> And why include the dot in the extension?

Just to be sure it's an actual file extension and not a file name `structpdb` with no real extension.

Reply:
> Since this is specific to pdb files, I thought a simple line was better than calling a function to differentiate different file extensions.

I don't know what you mean. `fileExtension()` simply returns all characters after the last dot in the string. It does exactly the same thing as your line, but avoid hard coding the length of the extension. Also avoids the check for the extension length, since it returns "invalid" when no file extension has been found.

```c++
if(fileExtension(getAtomFilepath()) == "pdb"){...}
```

Reply:
> I don't know what you mean. `fileExtension()` simply returns all characters after the last dot in the string. It does exactly the same thing as your line, but avoid hard coding the length of the extension. Also avoids the check for the extension length, since it returns "invalid" when no file extension has been found.

True, but `fileExtension()` is an auxiliary function of model_filereading.cpp
Do you want me to copy it as an auxiliary function in base_event.cpp or to make it publicly available somewhere else (such as in controller.cpp to be accessible by both the front-end and back-end)?

Reply:
Put the function in `misc.h` and you can use it everywhere

Reply:
Done, see merge request 41.